test/built-ins/TypedArray/prototype/slice/detached-buffer-get-custom-ctor.js

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.slice
+description: Throws a TypeError buffer is detached on Get custom constructor.
+info: >
+  22.2.3.24 %TypedArray%.prototype.slice ( start, end )
+
+  ...
+  9. Let A be ? TypedArraySpeciesCreate(O, « count »).
+  ...
+  14. If SameValue(srcType, targetType) is false, then
+    a. Let n be 0.
+    b. Repeat, while k < final
+      i. Let Pk be ! ToString(k).
+      ii. Let kValue be ? Get(O, Pk).
+  ...
+includes: [testTypedArray.js, detachArrayBuffer.js]
+features: [Symbol.species]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA();
+
+  sample.constructor = {};
+  sample.constructor[Symbol.species] = function(count) {
+    $DETACHBUFFER(sample.buffer);
+    return new TA(count);
+  };
+
+  assert.throws(TypeError, function() {
+    sample.slice();
```

Comment:
I believe this test is incorrect:

- step 3: len is 0
- step 4: relativeStart is 0
- step 5: k is 0
- step 6: relativeEnd is 0
- step 7: final is 0
- step 14.b: "k < final" is false

...meaning step 14.b.ii is never executed, so there should be no TypeError.